shuup/importer/_mapper.py

```diff
@@ -112,8 +113,10 @@ def _create_new_object(self, mapped, value):  # noqa (C901)
         if self.is_translated:
             obj.set_current_language(self.handler.language)
             for field in self.translated_fields:
-                if field.name not in ("master", "id", "language_code", "description"):
+                if field.name not in ("master", "id", "language_code", "description", "slug"):
                     setattr(obj, field.name, value)
+                elif field.name == "slug":
+                    setattr(obj, field.name, slugify(value[:field.max_length]))
```

Comment:
I guess you want to do this to all fields having "max_length" and not just slugfield. Also not sure if worth re-calling slugify here.

Reply:
I am not super expert with this relation mapper but I agree with @tulimaki . Making the max length check here is good but should happen for all fields that have this set, not only for "slug".
IMO, the slugify here depends on the import class as this is a "custom" behavior. Maybe not all importers want to slugify the value again?
Check https://github.com/shuup/shuup/blob/f68d268c72953e548ce0bae93fdced9793ecdb81/shuup/importer/importing/importing.py#L497
Maybe your importer can set this mapper to handle the relation the way you want instead of putting this custom logic inside Shuup OS?

Reply:
Or add way to customize this behavior in your importer.

Reply:
> Making the max length check here is good but should happen for all fields that have this set, not only for "slug".
I'll change it for all fields
> Maybe not all importers want to slugify the value again?
slugify value at this point sets the first time

Reply:
I guess the point here is that if have defined this slug-field in my excel like "Hello-World" this will make it always "hello-world" which is not ideal. You can override this in your custom product importer, but probably not good thing to do in this default implementation.

Reply:
removed